### PR TITLE
refactor: replace isPad function with isTablet 

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Noticed that we are using in most places a legacy implementation of a custom function to determine if a device is a tablet.

Replaced the old custom implementation with `isTablet()` function from `react-native-device-info` and updated some tests to reflect that.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- refactor: replace isPad function with isTablet  - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/utils/hardware.tests.ts [hunk 1]
```diff
@@ -1,71 +1,22 @@
+import RNDeviceInfo from "react-native-device-info"
+import { truncatedTextLimit } from "./hardware"
+
 jest.mock("react-native", () => ({
-  Platform: {
-    isPad: true,
-    OS: "ios",
-  },
```
Comment (gkartalis) [inline src/app/utils/hardware.tests.ts:5]: not needed anymore, also OS is by default iOS on jest

**Comment by gkartalis** [comment]
@brainbicycle curious if we can cleanup the `isPad` from the native code as well (pasted below a screenshot of the references of it)

Screenshot 2023-10-26 at 10 36 01

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (refactor: replace isPad function with isTablet  - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b4f210809ced037a1ac74f1260351f4b4c7bbf12
</p>

**Review by araujobarret — APPROVED** [review]
We might need to run a smoke test in a tablet just to be extra-safe before releasing it 👍

**Comment by gkartalis** [comment]
Totally agree, will handle that with a team QA possibly. Will merge after this release

**Review by MounirDhahri — APPROVED** [review]
Nice! thanks. Also true, we should probably keep the ones on the native side since they're supposedly being used

**Pull request merged**